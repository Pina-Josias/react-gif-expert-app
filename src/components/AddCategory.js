import React, { useState } from "react";
import PropTypes from "prop-types";

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputValue = (event) => {
    const { value } = event.target;
    setInputValue(value);
  };

  const findDuplicate = (elements) => elements.includes(inputValue);

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((elements) => {
        if (!findDuplicate(elements)) {
          return [inputValue, ...elements];
        }
        return elements;
      });
    }
  };
  return (
    <>
      <h2>Add Category</h2>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          value={inputValue}
          onChange={(event) => onInputValue(event)}
        />
      </form>
    </>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
