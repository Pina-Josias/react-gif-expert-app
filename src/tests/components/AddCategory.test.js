import React from "react";
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas sobre el componente de <AddCategory/>", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("debe mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", {
      target: {
        value: value,
      },
    });
    const p = wrapper.find("p");

    expect(p.text()).toBe(value);
  });

  test("no debe de post la informacion al hacer submit", () => {
    const form = wrapper.find("form");
    form.simulate("submit", {
      preventDefault() {},
    });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("debe llamarse el setCategory y limpiar la caja de texto", () => {
    const input = wrapper.find("input");
    input.simulate("change", {
      target: {
        value: "Hola Josias",
      },
    });
    const form = wrapper.find("form");
    form.simulate("submit", {
      preventDefault() {},
    });

    expect(setCategories).toHaveBeenCalled();

    expect(input.prop("value")).toBe("");
  });
});
