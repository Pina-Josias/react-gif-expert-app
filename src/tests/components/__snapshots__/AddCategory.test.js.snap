// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente de <AddCategory/> debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2>
    Add Category
  </h2>
  <p />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
